pkg/agent/cniserver/pod_configuration_linux_test.go

```diff
@@ -0,0 +1,514 @@
+	if c.configureContainerLinkError != nil {
+		return c.configureContainerLinkError
+	}
+	hostIface := &current.Interface{}
+	containerIface := &current.Interface{Name: containerIfaceName, Sandbox: containerNetNS}
+	if podSriovVFDeviceID != "" {
+		hostIface.Name = containerIfaceName
+	} else {
+		hostIface.Name = c.hostIfaceName
+		hostIface.Mac = hostIfaceMAC
+		containerIface.Mac = containerMAC
+	}
+	result.Interfaces = []*current.Interface{hostIface, containerIface}
+	return nil
+}
+
+func (c *fakeInterfaceConfigurator) removeContainerLink(containerID, hostInterfaceName string) error {
+	if c.removeContainerLinkError != nil {
+		return c.removeContainerLinkError
+	}
+	c.hostIfaceName = ""
+	return nil
+}
+
+func (c *fakeInterfaceConfigurator) advertiseContainerAddr(containerNetNS string, containerIfaceName string, result *current.Result) error {
+	return c.advertiseContainerAddrError
+}
+
+func (c *fakeInterfaceConfigurator) validateVFRepInterface(sriovVFDeviceID string) (string, error) {
+	return c.hostIfaceName, c.validateVFRepInterfaceError
+}
+
+func (c *fakeInterfaceConfigurator) validateContainerPeerInterface(interfaces []*current.Interface, containerVeth *vethPair) (*vethPair, error) {
+	return containerVeth, c.validateContainerPeerInterfaceError
+}
+
+func (c *fakeInterfaceConfigurator) getInterceptedInterfaces(sandbox string, containerNetNS string, containerIFDev string) (*current.Interface, *current.Interface, error) {
+	containerIface := &current.Interface{
+		Name:    containerIFDev,
+		Sandbox: sandbox,
+		Mac:     c.containerMAC,
+	}
+	hostIface := &current.Interface{
+		Name: c.hostIfaceName,
+		Mac:  c.hostIfaceMAC,
+	}
+	return containerIface, hostIface, c.getInterceptedInterfacesError
+}
+
+func (c *fakeInterfaceConfigurator) addPostInterfaceCreateHook(containerID, endpointName string, containerAccess *containerAccessArbitrator, hook postInterfaceCreateHook) error {
+	return nil
+}
+
+func (c *fakeInterfaceConfigurator) checkContainerInterface(containerNetns, containerID string, containerIface *current.Interface, containerIPs []*current.IPConfig, containerRoutes []*cnitypes.Route, sriovVFDeviceID string) (interface{}, error) {
+	if c.checkContainerInterfaceError != nil {
+		return nil, c.checkContainerInterfaceError
+	}
+	if sriovVFDeviceID != "" {
+		return c.containerVFLink, nil
+	}
+	return c.containerVethPair, nil
+}
+
+func newTestInterfaceConfigurator() *fakeInterfaceConfigurator {
+	return &fakeInterfaceConfigurator{
+		containerMAC: "01:02:03:04:05:06",
+		hostIfaceMAC: hostIfaceMAC,
+	}
+}
+
+func TestConnectInterceptedInterface(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+	defer func() {
+		getNSPath = util.GetNSPath
```

Comment:
please move it after where it's mocked for readability

Reply:
updated.